### PR TITLE
Modify LU_backward and lu_solve_backward to use linalg_solve_triangular

### PR DESCRIPTION
**Pull request by lezcano**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #63570
* __->__ #63569

This PR also rewrites `lu_solve_backward` from scratch going from
solving 5 systems of equations to just 2.

cc @jianyuh @nikitaved @pearu @mruberry @walterddr @IvanYashchuk @xwang233 @Lezcano

Differential Revision: [D32618014](https://our.internmc.facebook.com/intern/diff/D32618014)

### CODE REVIEW

**Review by nikitaved — APPROVED** [review]
LGTM! Thanks, Mario!

File: torch/csrc/autograd/FunctionsManual.cpp [hunk 1]
```diff
@@ -2857,19 +2857,15 @@ Tensor linalg_qr_backward(const std::vector<torch::autograd::Variable> &grads, c
...
-    // triangular_solve(rhs_term^H, R)^H
-    Tensor grad_A;
-    std::tie(grad_A, std::ignore) = at::triangular_solve(
-        rhs_term.mH(),
-        R,
-        /*upper=*/true,
-        /*transpose=*/false,
+    // Compute rhs_term @ R^{-H}
+    Tensor grad_A = at::linalg_solve_triangular(
+        R.transpose(-2, -1).conj(),
```
Comment (anjali411) [inline torch/csrc/autograd/FunctionsManual.cpp:2862]: can we just call `R.adjoint()` instead?

also looks like `adjoint`'s documentation doesn't render properly https://pytorch.org/docs/master/generated/torch.Tensor.adjoint.html#torch.Tensor.adjoint

**Review by anjali411 — APPROVED** [review]
Thank you! This is a cool change. It'd be nice to benchmark the performance.

**Comment by anjali411** [comment]
@anjali411 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32618014).

File: torch/csrc/autograd/FunctionsManual.cpp (hunk 1 above)
Comment (Lezcano) [reply torch/csrc/autograd/FunctionsManual.cpp:2862]: It renders the same as all the other tensor methods (for good or bad...)

And sure, I'll change it into using `mH()` once I go over this function again when moving `lu_solve` to `linalg.`.

Comment (anjali411) [reply torch/csrc/autograd/FunctionsManual.cpp:2862]: ok I meant I don't see docs for `torch.adjoint`. Probably missing an entry in some rst file.

**Review by mruberry — APPROVED** [review]
Cool!

File: torch/csrc/autograd/FunctionsManual.cpp (hunk 1 above)
Comment (mruberry) [reply torch/csrc/autograd/FunctionsManual.cpp:2862]: Would you file an issue?

Comment (Lezcano) [reply torch/csrc/autograd/FunctionsManual.cpp:2862]: oh, I had not realised that clicking on adjoint does not take you anywhere. I'll write a fix.

Comment (Lezcano) [reply torch/csrc/autograd/FunctionsManual.cpp:2862]: https://github.com/pytorch/pytorch/pull/68869

**Pull request closed without merging**